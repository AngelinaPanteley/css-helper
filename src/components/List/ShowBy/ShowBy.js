import React, { PureComponent } from 'react';

import styles from './ShowBy.scss';

class ShowBy extends PureComponent {
  render() {
    return (
      <div className={styles.Wrapper}>
        <p>Show by
          <select value={this.props.value}
            onChange={(e) => this.props.onChange(e.target.value)}>
            <option value={5}>5</option>
            <option value={10}>10</option>
            <option value={20}>20</option>
            <option value={50}>50</option>
          </select>
          /{this.props.length}
        </p>
      </div>
    )
  }
}

export default ShowBy;