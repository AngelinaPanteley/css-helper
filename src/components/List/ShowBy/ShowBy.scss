.Wrapper {
  color: #fff;
  font-size: 20px;
  text-align: right;
  margin-bottom: 20px;
}