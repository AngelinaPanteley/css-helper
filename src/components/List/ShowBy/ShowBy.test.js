"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';

import ShowBy from './ShowBy.js';

configure({ adapter: new Adapter() });

describe('>>>ShowBy', () => {

  //SNAPSHOTS

  const props = {
    value: '10',
    onChange: () => { },
    length: 20,
  };

  const showByComponent = <ShowBy {...props} />;

  const component = renderer.create(showByComponent);

  it('renders correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(showByComponent);
  });

  it('renders select', () => {
    chaiExpect(wrapper.find('select')).to.have.lengthOf(1);
  });
  it('renders two options', () => {
    chaiExpect(wrapper.find('option')).to.have.lengthOf(4);
  });
});
