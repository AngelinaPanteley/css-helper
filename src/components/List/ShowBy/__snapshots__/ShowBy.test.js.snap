// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ShowBy renders correctly 1`] = `
<div>
  <p>
    Show by
    <select
      onChange={[Function]}
      value="10"
    >
      <option
        value={5}
      >
        5
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={20}
      >
        20
      </option>
      <option
        value={50}
      >
        50
      </option>
    </select>
    /
    20
  </p>
</div>
`;
